### DISCUSSION TITLE
Account creation & OTP via JS extension

### DISCUSSION THREAD

**Discussion by SyahmiRafsan** [pocketbase/pocketbase#5913] [Q&A] [answered]
Hi, currently exploring the v0.23 OTP auth and just wondering if there is a JS extension equivalent for this:

`await pb.collection("users").authWithOTP(result.otpId, "EMAIL_CODE");`

I would like to create a `/register` endpoint so that I can:
1. Create a user account given email
2. Send user the OTP and logs in the user.

For subsequent logins, I will just use the client JS sdk code above.

Tried looking on the JSVM docs but I think that hasn't been updated yet for v0.23.

Thank you for the awesome project!

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not entirely sure that I understand the desired flow but if you use the latest v0.23.0-rc14 or the `develop` branch, then you can attach to the `OnRecordRequestOTPRequest` hook and assign your created user to `e.Record`. With the JS `pb_hooks` this could look something like (_I haven't tested it_):

```js
onRecordRequestOTPRequest((e) => {
    // create a user with the OTP email if it doesn't exist
    if (!e.record) {
        // read the email and any other extra user data you are submitting
        // with the OTP request to create a new user
        let email = e.requestInfo().body["email"]

        let record = new Record(e.collection)
        record.setEmail(email)
        record.setPassword($security.randomString(30))
        e.app.save(record)

        e.record = record
    }

    return e.next()
}, "users")
```

**Reply by SyahmiRafsan** [reply]
Thank you for this!

**Reply by SyahmiRafsan** [reply]
Is it possible to have a dynamic email for the OTP template?

For example, my app has 2 locales so for 'en' locale I will use the basic template as per the admin UI but I also have another locale where the email template will be different.

Edited: found the relevant hook which is `onMailerRecordOTPSend`

**Reply by einarpersson** [reply]
I'm looking for something similar.

I suppose you would also want some kind of cleanup of this temporary user if the OTP isn't verified within some timeframe?

**Reply by einarpersson** [reply]
Hmm could I use `onModelAfterDeleteSuccess` for the `otps` system collection for this? Is this called when a OTP is expired? 
(If so I could check if there is an unverified user and if so delete it)
